python/lsst/validate/drp/calcsrd/tex.py

```diff
@@ -0,0 +1,281 @@
+
+    Convenience function for calling correlation_function_ellipticity.
+
+    Parameters
+    ----------
+    matches : `lsst.validate.drp.matchreduce.MatchedMultiVisitDataset`
+        - The matched catalogs to analyze.
+
+    Returns
+    -------
+    r, xip, xip_err : each a np.array(dtype=float)
+        - The bin centers, two-point correlation, and uncertainty.
+    """
+    ra = matches.aggregate(averageRaFromCat) * u.radian
+    dec = matches.aggregate(averageDecFromCat) * u.radian
+
+    e1_res = matches.aggregate(medianEllipticity1ResidualsFromCat)
+    e2_res = matches.aggregate(medianEllipticity2ResidualsFromCat)
+
+    return correlation_function_ellipticity(ra, dec, e1_res, e2_res, **kwargs)
+
+
+def correlation_function_ellipticity(ra, dec, e1_res, e2_res,
+                                     nbins=20, min_sep=0.25, max_sep=20,
+                                     sep_units='arcmin', verbose=False):
+    """Compute shear-shear correlation function from ra, dec, g1, g2.
+
+    Default parameters for nbins, min_sep, max_sep chosen to cover
+       an appropriate range to calculate TE1 (<=1 arcmin) and TE2 (>=5 arcmin).
+    Parameters
+    ----------
+    ra : numpy.array
+        Right ascension of points [radians]
+    dec : numpy.array
+        Declination of points [radians]
+    e1_res : numpy.array
+        Residual ellipticity 1st component
+    e2_res : numpy.array
+        Residual ellipticity 2nd component
+    nbins : float, optional
+        Number of bins over which to analyze the two-point correlation
+    min_sep : float, optional
+        Minimum separation over which to analyze the two-point correlation
+    max_sep : float, optional
+        Maximum separation over which to analyze the two-point correlation
+    sep_units : str, optional
+        Specify the units of min_sep and max_sep
+    verbose : bool
+        Request verbose output from `treecorr`.
+        verbose=True will use verbose=2 for `treecorr.GGCorrelation`.
+
+    Returns
+    -------
+    r, xip, xip_err : each a np.array(dtype=float)
+        - The bin centers, two-point correlation, and uncertainty.
+    """
+    # Translate to 'verbose_level' here to refer to the integer levels in TreeCorr
+    # While 'verbose' is more generically what is being passed around
+    #   for verbosity within 'validate_drp'
+    if verbose:
+        verbose_level = 2
+    else:
+        verbose_level = 0
+
+    catTree = treecorr.Catalog(ra=ra, dec=dec, g1=e1_res, g2=e2_res,
+                               dec_units='radian', ra_units='radian')
+    gg = treecorr.GGCorrelation(nbins=nbins, min_sep=min_sep, max_sep=max_sep,
+                                sep_units=sep_units,
+                                verbose=verbose_level)
+    gg.process(catTree)
+    r = np.exp(gg.meanlogr) * u.arcmin
+    xip = gg.xip * u.Unit('')
+    xip_err = np.sqrt(gg.varxi) * u.Unit('')
+
+    return (r, xip, xip_err)
```

Comment:
Might be better to return a `namedtuple`, so the user doesn't have to keep track of which index is what thing?

Reply:
I don't disagree, but I've been generically discouraged from using `namedtuple` in favor of pipe_base struct.  But I don't want to invoke pipe_base struct here.